c7n/policy.py

```diff
@@ -480,6 +480,14 @@ def expand_variables(self, variables):
             if 'role' in variables['policy']['mode']:
                 mode = variables['policy']['mode'].copy()
                 mode['role'] = mode['role'].format(**variables)
+                if mode['use_host_account_id']:
```

Comment:
the line of code right above this effectively already supports this, ie. it substitutes the account id into the role. how is that different then what your doing here?

Reply:
fixed